### DISCUSSION TITLE
Support for isFetching/fetching

### DISCUSSION THREAD

**Discussion by chimon2000** [rrousselGit/riverpod#1608] [Ideas]
React Query & Redux Toolkit Query both have the concept of a `fetching` flag such that
* You can be fetching and be in a success(data) or error state at the same time.
* You cannot be in loading and success states at the same time.

The reasoning is to allow for greater control when rendering the UI: displaying a skeleton for `isLoading` vs. old data when `isFetching`: https://redux-toolkit.js.org/rtk-query/usage/queries#query-loading-state

Is this behavior possible with `FutureProvider` and `AsyncValue`, and if not whether it's worth exploring?

**Comment by TimWhiting** [comment]
See `isRefreshing`, in riverpod 2.0.0-dev's changelog: 
https://pub.dev/packages/riverpod/versions/2.0.0-dev.9/changelog#200-dev0

**Comment by rrousselGit** [comment]
2.0 does that

I'm making last minute changes on that behavior before releasing 2.0, but that new version is supposed to cover that.
